torchinductor/codegen/cpp.py

```diff
@@ -63,22 +63,24 @@ def cpp_prefix():
             #include <atomic>
             #include <cmath>
             #include <cstdlib>
-            #include <iostream>
             #include <limits>
-            #include <random>
             #include <omp.h>
 
+            #include <ATen/core/PhiloxRNGEngine.h>
+
             template<typename T>
             inline T mod(T a, T b) { return a % b; }
             template<>
             inline float mod(float a, float b) { return std::fmod(a, b); }
             template<>
             inline double mod(double a, double b) { return std::fmod(a, b); }
 
-            float rand_cpu(unsigned int seed) {
-                static thread_local std::mt19937 gen(seed ^ omp_get_thread_num());
-                static_assert(std::mt19937::min() == 0);
-                return gen() * static_cast<float>(1.0 / (std::mt19937::max() + 1.0));
+            constexpr float normalize (uint32_t value) {
+                return value * static_cast<float>(1.0 / std::numeric_limits<uint32_t>::max());
+            }
+
+            float normalized_rand_cpu(uint64_t seed, uint64_t offset) {
+                return normalize(at::Philox4_32_10(seed, omp_get_thread_num(), offset)());
```

Comment:
Why do we need `omp_get_thread_num()` here?  Deterministic random should not depend on openmp scheduler.